data/abi/abi_value.go

```diff
@@ -0,0 +1,304 @@
+func MakeUint16(value uint16) (Value, error) {
+	bigInt := big.NewInt(int64(value))
+	return MakeUint(bigInt, 16)
+}
+
+// MakeUint32 takes a go `uint32` and gives an ABI Value of ABI type `uint32`.
+func MakeUint32(value uint32) (Value, error) {
+	bigInt := big.NewInt(int64(value))
+	return MakeUint(bigInt, 32)
+}
+
+// MakeUint64 takes a go `uint64` and gives an ABI Value of ABI type `uint64`.
+func MakeUint64(value uint64) (Value, error) {
+	bigInt := big.NewInt(int64(0)).SetUint64(value)
+	return MakeUint(bigInt, 64)
+}
+
+// MakeUint takes a big integer representation and a type bitSize,
+// and returns an ABI Value of ABI Uint<bitSize> type.
+func MakeUint(value *big.Int, size uint16) (Value, error) {
+	typeUint, err := MakeUintType(size)
+	if err != nil {
+		return Value{}, err
+	}
+	upperLimit := big.NewInt(0).Lsh(big.NewInt(1), uint(size))
+	if value.Cmp(upperLimit) >= 0 {
+		return Value{}, fmt.Errorf("passed value larger than uint bitSize %d", size)
+	}
+	return Value{
+		ABIType: typeUint,
+		value:   value,
+	}, nil
+}
+
+// MakeUfixed takes a big rational number representation, a type bitSize, and a type precision,
+// and returns an ABI Value of ABI UFixed<bitSize>x<precision>
+func MakeUfixed(value *big.Int, size uint16, precision uint16) (Value, error) {
+	ufixedValueType, err := MakeUfixedType(size, precision)
+	if err != nil {
+		return Value{}, err
+	}
+	uintVal, err := MakeUint(value, size)
+	if err != nil {
+		return Value{}, err
+	}
+	uintVal.ABIType = ufixedValueType
+	return uintVal, nil
+}
+
+// MakeString takes a string and returns an ABI String type Value.
+func MakeString(value string) Value {
+	return Value{
+		ABIType: MakeStringType(),
+		value:   value,
+	}
+}
+
+// MakeByte takes a byte and returns an ABI Byte type value.
+func MakeByte(value byte) Value {
+	return Value{
+		ABIType: MakeByteType(),
+		value:   value,
+	}
+}
+
+// MakeAddress takes an [32]byte array and returns an ABI Address type value.
+func MakeAddress(value [32]byte) Value {
+	return Value{
+		ABIType: MakeAddressType(),
+		value:   value,
+	}
+}
+
+// MakeDynamicArray takes an array of ABI value (can be empty) and element type,
+// returns an ABI dynamic length array value.
+func MakeDynamicArray(values []Value, elemType Type) (Value, error) {
+	if len(values) >= (1 << 16) {
+		return Value{}, fmt.Errorf("dynamic array make error: pass in argument number larger than 2^16")
+	}
+	for i := 0; i < len(values); i++ {
+		if !values[i].ABIType.Equal(elemType) {
+			return Value{}, fmt.Errorf("type mismatch: %s and %s",
+				values[i].ABIType.String(), elemType.String())
+		}
+	}
+	return Value{
+		ABIType: MakeDynamicArrayType(elemType),
+		value:   values,
+	}, nil
+}
+
+// MakeStaticArray takes an array of ABI value and returns an ABI static length array value.
+func MakeStaticArray(values []Value) (Value, error) {
+	if len(values) >= (1 << 16) {
+		return Value{}, fmt.Errorf("static array make error: pass in argument number larger than 2^16")
```

Comment:
Same constant comment, and this error message should talk about length of array, not "argument number".